packages/main/src/MenuItem.ts

```diff
@@ -382,6 +404,72 @@ class MenuItem extends ListItem implements IMenuItem {
 		return this.items.filter((item): item is MenuItem => !item.isSeparator);
 	}
 
+	_closeOtherSubMenus(item: MenuItem) {
+		const menuItems = this._menuItems;
+		if (menuItems.indexOf(item) < 0) {
+			return;
+		}
+
+		menuItems.forEach(menuItem => {
+			if (menuItem !== item) {
+				menuItem._close();
+			}
+		});
+	}
+
+	_itemMouseOver(e: MouseEvent) {
+		if (!isDesktop()) {
+			return;
+		}
+		const item = e.target as MenuItem;
+
+		if (!MenuItem._isInstanceOfMenuItem(item)) {
+			return;
+		}
+		item.focus();
+
+		this._closeOtherSubMenus(item);
+	}
+
+	_itemKeyDown(e: KeyboardEvent) {
+		const item = e.target as MenuItem;
+		const itemInMenuItems = this._menuItems.indexOf(item) > -1;
+		const isTabNextPrevious = isTabNext(e) || isTabPrevious(e);
+		const isItemNavigation = isUp(e) || isDown(e);
+		const isItemSelection = isSpace(e) || isEnter(e);
+		const shouldOpenMenu = this.isRtl ? isLeft(e) : isRight(e);
+		const shouldCloseMenu = !(isItemNavigation || isItemSelection || shouldOpenMenu) || isTabNextPrevious;
+
+		if (itemInMenuItems && shouldCloseMenu) {
```

Comment:
This is something minor, and my opinion, but this looks better to me (you can ignore it if you want):

```ts
if (!itemInMenuItems || !shouldCloseMenu) {
    return;
}

this._close();
this.focus();
e.stopPropagation();
```